src/libraries/Common/src/Interop/Android/System.Security.Cryptography.Native.Android/Interop.EcDsa.ImportExport.cs

```diff
@@ -0,0 +1,334 @@
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyCreateByKeyParameters", CharSet = CharSet.Ansi)]
+        private static extern int EcKeyCreateByKeyParameters(
+            out SafeEcKeyHandle key,
+            string oid,
+            byte[]? qx, int qxLength,
+            byte[]? qy, int qyLength,
+            byte[]? d, int dLength);
+
+        internal static SafeEcKeyHandle EcKeyCreateByKeyParameters(
+            string oid,
+            byte[]? qx, int qxLength,
+            byte[]? qy, int qyLength,
+            byte[]? d, int dLength)
+        {
+            SafeEcKeyHandle key;
+            int rc = EcKeyCreateByKeyParameters(out key, oid, qx, qxLength, qy, qyLength, d, dLength);
+            if (rc == -1)
+            {
+                key?.Dispose();
+                Interop.Crypto.ErrClearError();
+
+                throw new PlatformNotSupportedException(SR.Format(SR.Cryptography_CurveNotSupported, oid));
+            }
+            return key;
+        }
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "CryptoNative_EcKeyCreateByExplicitParameters")]
+        internal static extern SafeEcKeyHandle EcKeyCreateByExplicitParameters(
+            ECCurve.ECCurveType curveType,
+            byte[]? qx, int qxLength,
+            byte[]? qy, int qyLength,
+            byte[]? d, int dLength,
+            byte[] p, int pLength,
+            byte[] a, int aLength,
+            byte[] b, int bLength,
+            byte[] gx, int gxLength,
+            byte[] gy, int gyLength,
+            byte[] order, int nLength,
+            byte[]? cofactor, int cofactorLength,
+            byte[]? seed, int seedLength);
+
+        internal static SafeEcKeyHandle EcKeyCreateByExplicitCurve(ECCurve curve)
+        {
+            byte[] p;
+            if (curve.IsPrime)
+            {
+                p = curve.Prime!;
+            }
+            else if (curve.IsCharacteristic2)
+            {
+                p = curve.Polynomial!;
+            }
+            else
+            {
+                throw new PlatformNotSupportedException(SR.Format(SR.Cryptography_CurveNotSupported, curve.CurveType.ToString()));
+            }
+
+            SafeEcKeyHandle key = Interop.Crypto.EcKeyCreateByExplicitParameters(
+                curve.CurveType,
+                null, 0,
+                null, 0,
+                null, 0,
+                p, p.Length,
+                curve.A!, curve.A!.Length,
+                curve.B!, curve.B!.Length,
+                curve.G.X!, curve.G.X!.Length,
+                curve.G.Y!, curve.G.Y!.Length,
+                curve.Order!, curve.Order!.Length,
+                curve.Cofactor, curve.Cofactor!.Length,
+                curve.Seed, curve.Seed == null ? 0 : curve.Seed.Length);
+
+            if (key == null || key.IsInvalid)
+            {
+                if (key != null)
+                    key.Dispose();
+                throw Interop.Crypto.CreateOpenSslCryptographicException();
```

Comment:
I think we just want a normal `CryptographicException` instead of `OpenSslCryptographicException`.

Reply:
Actually, I think this will actually throw a `CryptographicException` with AndroidCrypto - but with an extra P/Invoke to get there.